exercises/react/src/lib.rs

```diff
@@ -0,0 +1,94 @@
+    // You may return an Err (and you can change the error type) if the dependencies don't exist.
+    // This error behavior will not be tested, because it's not possible to test that without
+    // knowing what type you use for CellID.
+    //
+    // Notice that there is no way to *remove* a cell.
+    // This means that you may assume, without checking, that if the dependencies exist at creation
+    // time they will continue to exist as long as the Reactor exists.
+    pub fn create_compute<F: Fn(&[T]) -> T + 'a>(&mut self, dependencies: &[CellID], compute_func: F) -> Result<CellID, ()> {
+        unimplemented!()
+    }
+
+    // Retrieves the current value of the cell, or None if the cell does not exist.
+    //
+    // You may wonder whether it is possible to implement `get(&self, id: CellID) -> &Cell` and
+    // have a `value(&self)` method on `Cell`.
+    //
+    // It turns out this introduces a significant amount of extra complexity to this exercise.
+    // We chose not to cover this here, since this exercise is probably enough work as-is.
+    pub fn value(&self, id: CellID) -> Option<T> {
+        unimplemented!()
+    }
+
+    // Sets the value of the input cell.
+    //
+    // Return an Err (and you can change the error type) if the cell does not exist, or the
+    // specified cell is a compute cell (if that is possible in your implementation), since compute
+    // cells cannot have their values directly set.
+    //
+    // Similarly, you may wonder if we can have `get_mut(&mut self, id: CellID) -> &mut Cell`, with
+    // a `set_value(&mut self, new_value: T)` method on `Cell`.
+    //
+    // As before, that turned out to add too much extra complexity.
+    pub fn set_value(&mut self, id: CellID, new_value: T) -> Result<(), ()> {
+        unimplemented!()
+    }
+
+    // Adds a callback to the compute cell.
+    //
+    // Return an Err (and you can change the error type) if the cell does not exist.
+    //
+    // Callbacks on input cells (if even possible in your implementation) will not be tested.
+    //
+    // The semantics of callbacks (as will be tested):
+    // For a single set_value call, each callback should be called:
+    // * Zero times if the compute cell's value did not change as a result of the set_value call.
+    // * Exactly once if the compute cell's value changed as a result of the set_value call.
+    //   The value passed to the callback should be the final value of the compute cell after the
+    //   set_value call.
+    pub fn add_callback<F: FnMut(T) -> () + 'a>(&mut self, id: CellID, callback: F) -> Result<CallbackID, ()> {
+        unimplemented!()
+    }
+
+    // Removes the specified callback, using an ID returned from add_callback.
+    //
+    // Return an Err (and you can change the error type) if the cell does not exist.
+    //
+    // If the callback does not exist (or has already been removed), it is up to you whether to
+    // return Ok(()) or an error, as it cannot be tested.
```

Comment:
not true. repeatedly calling remove on the same callback ID can test this. Only question is whether we should:

* Test it and expect `Ok(())`
* Test it and expect `is_err()`
* Not test it.

In any case, change the comment.